torch/_inductor/ir.py

```diff
@@ -684,6 +684,30 @@ def create(
         reduction_hint: ReductionHint = ReductionHint.DEFAULT,
     ):
         reduction_numel = V.graph.sizevars.simplify(sympy_product(reduction_ranges))
+
+        if reduction_numel == 0:
+
+            rtypes_to_inits = {
+                "sum": 0,
+                "prod": 1,
+                "any": 0,
```

Comment:
there's no "any" rtype

Reply:
Did you mean `all`? I put a print in that function and I'm seeing 
```
@@@2222reduction_type=any
```

making it into this function.

```
        if reduction_numel == 0:

            print(f"@@@2222reduction_type={reduction_type}")
            rtypes_to_inits = {
                "sum": 0,
                "prod": 1,
                "any": 0,
                "all": 0,
            }
```

Reply:
Right, there's `any`, there's no `all`. We should make rtype an enum and not string.

Reply:
right, I'll do the enum thing in a follow-up PR.